guides/07_streaming/03_object-detection-from-video.md

```diff
@@ -0,0 +1,186 @@
+# Object Detection from a Webcam Stream
+
+Tags: VISION, STREAMING, VIDEO
+
+In this guide we'll use the [RT-DETR](https://huggingface.co/docs/transformers/en/model_doc/rt_detr) model to detect objects in a user uploaded video. We'll stream the results from the server using the new video streaming  
+features introduced in Gradio 5.0.
+
+## Setting up the Model
+
+First, we'll install the following requirements in our system:
+
+```
+opencv-python
+torch
+transformers
+spaces
+```
+
+Then, we'll download the model from the Hugging Face Hub:
+
+```python
+from transformers import RTDetrForObjectDetection, RTDetrImageProcessor
+
+image_processor = RTDetrImageProcessor.from_pretrained("PekingU/rtdetr_r50vd")
+model = RTDetrForObjectDetection.from_pretrained("PekingU/rtdetr_r50vd").to("cuda")
+```
+
+We're moving the model to the GPU. We'll be deploying our model to Hugging Face Spaces and running the inference in the free ZeroGPU cluster. 
+
+
+## The Inference Function
+
+Our inference function will accept a video and a desired confidence threshold.
+Object detection models identify many objects and assign a confidence score to each object. The lower the confidence, the higher the chance of a false positive. So we will let our users set the conference threshold.
+
+Our function will iterate over the frames in the video and run the RT-DETR model over each frame.
+We will then draw the bounding boxes for each detected object in the frame and save the frame to a new output video.
+The function will yield each output video in chunks of two seconds.
+
+In order to keep inference times as low as possible on ZeroGPU (there is a time-based quota),
+we will halve the original frames-per-second in the output video and resize the input frames to be half the original 
+size before running the model.
```

Comment:
```suggestion
size before running the model. If you are running on your GPU, you can set  SUBSAMPLE to 1 in the code below.
```
sug: you might want to make the halving of the width/height easy to configure in case people have a beefy gpu they want to use